columns/Column.js

```diff
@@ -140,7 +144,7 @@ export class Column {
             field: this.field,
             colId: this.colId,
             headerName: this.headerName,
-            hide: this.hide,
+            hide: this.hide && this.hideable,
```

Comment:
This is questionable, but I thought we show not only enforce this prop via the col chooser but protect also from programatic changes (possibly from state?). This way ag-grid never gets a hide property on an unhideable col.

I could see this behavior confusing an app developer if they intentionally try to hide an 'unhideable' col but it still pops up in the grid. On the other hand why would you try to do such a thing?

Reply:
This one is a toss-up and I could see pros and cons either way

Reply:
Thinking about this some more, I think its easiest to reason about if hide solely determines the visibility, and 'hideable' just governs the ability to hide it in the column chooser.  Also seems analogous to 'resizable' and 'moveable' in that it just governs user initiated GUI actions.